### ISSUE TITLE
Three test_mainwindow.py test failures: test_runcell_after_restart, test_varexp_cleared_after_kernel_restart and test_print_faulthandler

### ISSUE THREAD

**Issue by juliangilbey** [spyder-ide/spyder#17278]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

Following on from #17276, there are three new tests failing in the `--run-slow` tests (and besides `test_get_pydoc`, `test_qtbug35861` and the enigmatic #17103, that is it for test suite failures). These three are in test_mainwindow.py (using python3-ipython version 7.31.1-1). Here is the test output.

```
__________________________ test_runcell_after_restart __________________________

main_window = <spyder.app.mainwindow.MainWindow object at 0x7f0268383010>
qtbot = <pytestqt.qtbot.QtBot object at 0x7f0364f2d4b0>

    @pytest.mark.slow
    @flaky(max_runs=3)
    @pytest.mark.skipif(sys.platform == 'darwin',
                        reason="Fails sometimes on macOS")
    def test_runcell_after_restart(main_window, qtbot):
        """Test runcell after a kernel restart."""
        # Write code to a file
        code = "print('test_runcell_after_restart')"
        # Wait until the window is fully up
        shell = main_window.ipyconsole.get_current_shellwidget()
        qtbot.waitUntil(lambda: shell._prompt_html is not None,
                        timeout=SHELL_TIMEOUT)
        # create new file
        main_window.editor.new()
        code_editor = main_window.editor.get_focus_widget()
        code_editor.set_text(code)
    
        # Restart Kernel
>       with qtbot.waitSignal(shell.sig_prompt_ready, timeout=10000):
E       pytestqt.exceptions.TimeoutError: Signal sig_prompt_ready() not emitted after 10000 ms

/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/app/tests/test_mainwindow.py:3417: TimeoutError
----------------------------- Captured Qt messages -----------------------------
QtWarningMsg: Attribute Qt::AA_UseSoftwareOpenGL must be set before QCoreApplication is created.
QtWarningMsg: Scenegraph already initialized, setBackend() request ignored
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
---------------------------- Captured stderr setup -----------------------------
<class 'spyder_boilerplate.spyder.plugin.SpyderBoilerplate'>: A plugin with section "spyder_boilerplate" already exists!
Traceback (most recent call last):
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/app/mainwindow.py", line 902, in setup
    plugin_instance = PLUGIN_REGISTRY.register_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 342, in register_plugin
    instance = self._instantiate_spyder5_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 169, in _instantiate_spyder5_plugin
    CONF.register_plugin(PluginClass)
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/config/manager.py", line 129, in register_plugin
    raise RuntimeError('A plugin with section "{}" already '
RuntimeError: A plugin with section "spyder_boilerplate" already exists!
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
---------------------------- Captured stderr setup -----------------------------
<class 'spyder_boilerplate.spyder.plugin.SpyderBoilerplate'>: A plugin with section "spyder_boilerplate" already exists!
Traceback (most recent call last):
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/app/mainwindow.py", line 902, in setup
    plugin_instance = PLUGIN_REGISTRY.register_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 342, in register_plugin
    instance = self._instantiate_spyder5_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 169, in _instantiate_spyder5_plugin
    CONF.register_plugin(PluginClass)
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/config/manager.py", line 129, in register_plugin
    raise RuntimeError('A plugin with section "{}" already '
RuntimeError: A plugin with section "spyder_boilerplate" already exists!
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
--------------------------- Captured stdout teardown ---------------------------
[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

In [1]: 
___________________ test_varexp_cleared_after_kernel_restart ___________________

main_window = <spyder.app.mainwindow.MainWindow object at 0x7f02282adfc0>
qtbot = <pytestqt.qtbot.QtBot object at 0x7f01ec318880>

    @pytest.mark.slow
    @flaky(max_runs=3)
    def test_varexp_cleared_after_kernel_restart(main_window, qtbot):
        """
        Test that the variable explorer is cleared after a kernel restart.
        """
        shell = main_window.ipyconsole.get_current_shellwidget()
        qtbot.waitUntil(lambda: shell._prompt_html is not None,
                        timeout=SHELL_TIMEOUT)
    
        # Create a variable
        with qtbot.waitSignal(shell.executed):
            shell.execute('a = 10')
    
        # Assert the value is shown in the variable explorer
        nsb = main_window.variableexplorer.current_widget()
        qtbot.waitUntil(lambda: 'a' in nsb.editor.source_model._data,
                        timeout=3000)
    
        # Restart Kernel
>       with qtbot.waitSignal(shell.sig_prompt_ready, timeout=10000):
E       pytestqt.exceptions.TimeoutError: Signal sig_prompt_ready() not emitted after 10000 ms

/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/app/tests/test_mainwindow.py:3618: TimeoutError
----------------------------- Captured Qt messages -----------------------------
QtWarningMsg: Attribute Qt::AA_UseSoftwareOpenGL must be set before QCoreApplication is created.
QtWarningMsg: Scenegraph already initialized, setBackend() request ignored
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: a = 10

In [2]: 
---------------------------- Captured stderr setup -----------------------------
<class 'spyder_boilerplate.spyder.plugin.SpyderBoilerplate'>: A plugin with section "spyder_boilerplate" already exists!
Traceback (most recent call last):
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/app/mainwindow.py", line 902, in setup
    plugin_instance = PLUGIN_REGISTRY.register_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 342, in register_plugin
    instance = self._instantiate_spyder5_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 169, in _instantiate_spyder5_plugin
    CONF.register_plugin(PluginClass)
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/config/manager.py", line 129, in register_plugin
    raise RuntimeError('A plugin with section "{}" already '
RuntimeError: A plugin with section "spyder_boilerplate" already exists!
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: a = 10

In [2]: 
---------------------------- Captured stderr setup -----------------------------
<class 'spyder_boilerplate.spyder.plugin.SpyderBoilerplate'>: A plugin with section "spyder_boilerplate" already exists!
Traceback (most recent call last):
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/app/mainwindow.py", line 902, in setup
    plugin_instance = PLUGIN_REGISTRY.register_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 342, in register_plugin
    instance = self._instantiate_spyder5_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 169, in _instantiate_spyder5_plugin
    CONF.register_plugin(PluginClass)
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/config/manager.py", line 129, in register_plugin
    raise RuntimeError('A plugin with section "{}" already '
RuntimeError: A plugin with section "spyder_boilerplate" already exists!
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: a = 10

In [2]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: a = 10

In [2]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: a = 10

In [2]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: a = 10

In [2]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: a = 10

In [2]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: a = 10

In [2]: 
___________________________ test_print_faulthandler ____________________________

main_window = <spyder.app.mainwindow.MainWindow object at 0x7f012c3eb370>
qtbot = <pytestqt.qtbot.QtBot object at 0x7f00bc3ebf70>
tmpdir = local('/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler2')

    @pytest.mark.slow
    @flaky(max_runs=3)
    @pytest.mark.skipif(
        os.name == 'nt',
        reason="ctypes.string_at(0) doesn't segfaults on Windows")
    def test_print_faulthandler(main_window, qtbot, tmpdir):
        """Test printing segfault info from kernel crashes."""
        # Write code with a cell to a file
        code = """
    def crash_func():
        import ctypes; ctypes.string_at(0)
    crash_func()
    """
    
        p = tmpdir.join("print-test.py")
        p.write(code)
        main_window.editor.load(to_text_string(p))
        shell = main_window.ipyconsole.get_current_shellwidget()
        qtbot.waitUntil(lambda: shell._prompt_html is not None,
                        timeout=SHELL_TIMEOUT)
        control = main_window.ipyconsole.get_widget().get_focus_widget()
    
        # Click the run button
        run_action = main_window.run_toolbar_actions[0]
        run_button = main_window.run_toolbar.widgetForAction(run_action)
        qtbot.mouseClick(run_button, Qt.LeftButton)
        qtbot.wait(5000)
    
>       assert 'Segmentation fault' in control.toPlainText()
E       assert 'Segmentation fault' in 'Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]\nType "copyright", "credits" or "license" for more informati...nt_faulthandler2\')\n\n[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled\n\nIn [1]: '
E        +  where 'Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]\nType "copyright", "credits" or "license" for more informati...nt_faulthandler2\')\n\n[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled\n\nIn [1]: ' = <built-in method toPlainText of ControlWidget object at 0x7f00f431a320>()
E        +    where <built-in method toPlainText of ControlWidget object at 0x7f00f431a320> = <spyder.plugins.ipythonconsole.widgets.control.ControlWidget object at 0x7f00f431a320>.toPlainText

/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/app/tests/test_mainwindow.py:4310: AssertionError
----------------------------- Captured Qt messages -----------------------------
QtWarningMsg: Attribute Qt::AA_UseSoftwareOpenGL must be set before QCoreApplication is created.
QtWarningMsg: Scenegraph already initialized, setBackend() request ignored
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

In [1]: runfile('/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler0/print-test.py', wdir='/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler0')

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
---------------------------- Captured stderr setup -----------------------------
<class 'spyder_boilerplate.spyder.plugin.SpyderBoilerplate'>: A plugin with section "spyder_boilerplate" already exists!
Traceback (most recent call last):
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/app/mainwindow.py", line 902, in setup
    plugin_instance = PLUGIN_REGISTRY.register_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 342, in register_plugin
    instance = self._instantiate_spyder5_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 169, in _instantiate_spyder5_plugin
    CONF.register_plugin(PluginClass)
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/config/manager.py", line 129, in register_plugin
    raise RuntimeError('A plugin with section "{}" already '
RuntimeError: A plugin with section "spyder_boilerplate" already exists!
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: runfile('/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler1/print-test.py', wdir='/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler1')

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
---------------------------- Captured stderr setup -----------------------------
<class 'spyder_boilerplate.spyder.plugin.SpyderBoilerplate'>: A plugin with section "spyder_boilerplate" already exists!
Traceback (most recent call last):
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/app/mainwindow.py", line 902, in setup
    plugin_instance = PLUGIN_REGISTRY.register_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 342, in register_plugin
    instance = self._instantiate_spyder5_plugin(
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/api/plugin_registration/registry.py", line 169, in _instantiate_spyder5_plugin
    CONF.register_plugin(PluginClass)
  File "/tmp/autopkgtest.VGygWg/build.wdI/src/spyder/config/manager.py", line 129, in register_plugin
    raise RuntimeError('A plugin with section "{}" already '
RuntimeError: A plugin with section "spyder_boilerplate" already exists!
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

In [1]: runfile('/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler0/print-test.py', wdir='/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler0')

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

In [1]: runfile('/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler0/print-test.py', wdir='/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler0')

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: runfile('/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler1/print-test.py', wdir='/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler1')

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

In [1]: runfile('/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler0/print-test.py', wdir='/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler0')

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: runfile('/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler1/print-test.py', wdir='/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler1')

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
--------------------------- Captured stdout teardown ---------------------------
Python 3.10.2 (main, Jan 16 2022, 17:11:27) [GCC 11.2.0]
Type "copyright", "credits" or "license" for more information.

IPython 7.31.1 -- An enhanced Interactive Python.

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: runfile('/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler2/print-test.py', wdir='/tmp/pytest-of-jdg/pytest-1/test_print_faulthandler2')

[SpyderKernelApp] WARNING | debugpy_stream undefined, debugging will not be enabled

In [1]: 
```

### What steps reproduce the problem?

Running the test suite in a Debian unstable chroot with `--run-slow`.

## Versions

* Spyder version: 5.2.2
* Python version: 3.9 and 3.10
* Qt version: 5.15.8+dfsg-1
* PyQt version: 5.15.6+dfsg-1+b1
* Operating System name/version: Debian unstable

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.0.2 (OK)
IPython >=7.6.0;<8.0.0        :  7.31.1 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.0 (OK)
jellyfish >=0.7               :  0.8.9 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.5.0 (OK)
nbconvert >=4.0               :  6.4.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.1 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.7.1 (OK)
pylint >=2.5.0                :  2.12.2 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.3.2;<1.4.0          :  1.3.3 (OK)
pylsp_black >=1.0.0           :  1.0.1 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.1.1 (OK)
qtconsole >=5.2.1;<5.3.0      :  5.2.2 (OK)
qtpy >=1.5.0                  :  2.0.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  59.6.0 (OK)
sphinx >=0.6.6                :  4.3.2 (OK)
spyder_kernels >=2.2.1;<2.3.0 :  2.2.1 (OK)
textdistance >=4.2.0          :  4.2.2 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
xdg >=0.26                    :  0.27 (OK)
zmq >=17                      :  22.3.0 (OK)

# Optional:
cython >=0.21                 :  0.29.24 (OK)
matplotlib >=2.0.0            :  3.5.1 (OK)
numpy >=1.7                   :  1.21.5 (OK)
pandas >=1.1.1                :  1.3.5 (OK)
scipy >=0.17.0                :  1.7.3 (OK)
sympy >=0.7.3                 :  1.7.1 (OK)
```
Best wishes, Julian

**Comment by ccordoba12** [comment]
I can't reproduce thiese failures, sorry. This is probably due to a compilation or packaging issue in your side.

**Comment by juliangilbey** [comment]
Thanks for the quick response!  As I wrote in #17276: that's really weird; my local setup is attempting to follow your upstream version as closely as possible.  The only differences (which are admittedly big ones) are the versions of PyQt5 and Qt5.  So maybe it's there, and things will get resolved once we get (closer) to version 5.3.0?  Happy to leave this open for now; I certainly don't see any obviously strange behaviour when running Spyder in this environment.

**Comment by ccordoba12** [comment]
Closing because I don't think this is a problem in our side.

**Comment by juliangilbey** [comment]
Yes, indeed.  I've had a little time to track down the cause, but with no success so far.  It definitely appears to be something to do with the environment.  When I succeed, I will let you know if it turns out to be something worth telling you about!

I have managed to track down the cause of the debugpy messages: it turns out that we don't have debugpy in Debian yet!  So I'm working on that one.
